src/todt.c

```diff
@@ -62,15 +60,15 @@ Symbol *toSymbolCppTypeInfo(ClassDeclaration *cd);
 
 /* ================================================================ */
 
-dt_t **Initializer_toDt(Initializer *init, dt_t **pdt)
+void Initializer_toDt(Initializer *init, DtBuilder& dtb)
 {
     class InitToDt : public Visitor
     {
     public:
-        dt_t **pdt;
+        DtBuilder dtb;
 
-        InitToDt(dt_t **pdt)
-            : pdt(pdt)
+        InitToDt(DtBuilder& dtb)
+            : dtb(dtb)
```

Comment:
Ah, OK I noticed. the `InitToDt.dtb.pTail` is copied from `dtb`, then it still points the original `dtb.head`. So it doesn't have a problem.
But... it looks still a bit weird. `DtBuilder` is designed to have interior pointer. I thought it meant, from my D experience, `DtBuilder` should not be copied. But it was my wrong assumption.

Can you make the code a bit cleaner by either way?

1. Add explicit copy constructor to DtBuilder, something like:
 ```cpp
 // This copied DtBuilder refers original dt structure.
 DtBuilder(DtBuilder& dtb) { head = dtb.head; pTail = dtb.pTail; }
 ```

2. Make the visitor's dtb member fields references.
 ```cpp
 class InitToDt : public Visitor { DtBuilder& dtb; ...};
 ```